app/src/main/java/tech/relaycorp/courier/domain/CollectPublicCargo.kt

```diff
@@ -0,0 +1,49 @@
+package tech.relaycorp.courier.domain
+
+import kotlinx.coroutines.flow.collect
+import tech.relaycorp.courier.data.database.StoredMessageDao
+import tech.relaycorp.courier.data.disk.DiskRepository
+import tech.relaycorp.courier.data.model.MessageAddress
+import tech.relaycorp.courier.data.model.MessageType
+import tech.relaycorp.courier.data.model.StoredMessage
+import tech.relaycorp.courier.data.network.CogRPC
+import javax.inject.Inject
+
+class CollectPublicCargo
+@Inject constructor(
+    private val storedMessageDao: StoredMessageDao,
+    private val cogRPC: CogRPC,
+    private val storeMessage: StoreMessage,
+    private val deleteMessage: DeleteMessage,
+    private val diskRepository: DiskRepository
+) {
+
+    suspend fun collect() {
+        getCCAs()
+            .forEach { cca ->
+                cogRPC
+                    .collectCargo(cca.recipientAddress.value, cca.toCogRPCMessage())
+                    .collect {
+                        storeMessage.storeCargo(
+                            cca.recipientAddress,
+                            MessageAddress.Type.Private,
+                            it.data
+                        )
```

Comment:
We should validate the cargo before we store it:

```kotlin
val cargo = Cargo.deserialize(it.data)
cargo.validate()
```

Note that the `.collectCargo()` method has to return the serialisation of the message, as opposed to the parsed message, because we wouldn't be able to reconstruct the original byte sequence.